backend/src/main/java/com/ziwg/reservatio/controllers/EmployeeController.java

```diff
@@ -32,41 +31,74 @@ public EmployeeController(EmployeeRepository employeeRepository,
-        List<String> serviceProviderServices = serviceProvider.get().getServices().stream().map(Service::getName).collect(Collectors.toList());
-        if (!serviceProviderServices.containsAll(employeePojo.getServices()))
+        Employee employee = createEmployee(employeePojo, serviceProvider.get());
+        List<String> serviceProviderServicesNames = serviceProvider.get().getServices().stream().map(Service::getName).collect(Collectors.toList());
+        if (!serviceProviderServicesNames.containsAll(employeePojo.getServices()))
             return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
-        List<Service> servicesToAdd = serviceProvider.get().getServices().stream().filter(service -> employeePojo.getServices().contains(service.getName())).collect(Collectors.toList());
-        employee.setServices(servicesToAdd);
+        setServicesForEmployee(employeePojo, employee);
         employeeRepository.save(employee);
         return new ResponseEntity<>(HttpStatus.CREATED);
     }
 
-    @DeleteMapping("deleteEmployee/{employeeId}")
-    public ResponseEntity<String> deleteEmployee(@PathVariable Long employeeId) {
-        Optional<Employee> employeeToDelete = employeeRepository.findById(employeeId);
+    private Employee createEmployee(EmployeePojo employeePojo, ServiceProvider serviceProvider) {
+        return Employee.builder().firstName(employeePojo.getFirstName()).lastName(employeePojo.getLastName()).serviceProvider(serviceProvider).build();
+    }
+
+    private void setServicesForEmployee(EmployeePojo employeePojo, Employee employee) {
+        List<Service> servicesToAdd = getServices(employeePojo, employee.getServiceProvider());
+        employee.setServices(servicesToAdd);
+    }
+
+    private List<Service> getServices(EmployeePojo employeePojo, ServiceProvider serviceProvider) {
+        return serviceProvider.getServices().stream().filter(service -> employeePojo.getServices().contains(service.getName())).collect(Collectors.toList());
+    }
+
+    @DeleteMapping("{employeeId}")
+    public ResponseEntity<String> deleteEmployee(@PathVariable Long serviceProviderId, @PathVariable Long employeeId) {
+        Optional<ServiceProvider> serviceProvider = serviceProviderRepository.findById(serviceProviderId);
+        if (serviceProvider.isEmpty())
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        Optional<Employee> employeeToDelete = getEmployeeFromServiceProvider(employeeId, serviceProvider.get());
         if (employeeToDelete.isEmpty())
             return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
-        if (employeeToDelete.get().getReservations().stream().anyMatch(reservation -> reservation.getDateTime().isAfter(LocalDateTime.now())))
+        if (hasAnyUpcomingReservations(employeeToDelete.get()))
             return new ResponseEntity<>("Cannot delete an employee that has upcoming appointments", HttpStatus.BAD_REQUEST);
```

Comment:
Przydałoby się bardziej konsekwentnie podejść do body zwracanego w przypadku błędów. To jest w sumie jedyne miejsce, gdzie zwracamy informację co poszło nie tak. Powinna ona być chyba zwracana zawsze.